qiskit_experiments/randomized_benchmarking/base_rb_generator.py

```diff
@@ -0,0 +1,332 @@
+            and B is the number of the seed.
+
+            When the underlying group is CNOT-Dihedral and not Clifford, the format is
+            "rb_cnotdihedral_C_length_A_seed_B"
+            Where C is either "X" or "Z", depending on the measurement basis used in the circuit
+
+            For non-standard types of RB experiments,
+            an additional TYPE string is part of the format:
+            "rb_TYPE_length_A_seed_B"
+        """
+        meta = circuit.metadata
+        name = "rb_"
+        if meta['group_type'] == 'cnot_dihedral':
+            name += "cnotdihedral_{}_".format(meta['cnot_basis'])
+        if self.circuit_type_string(meta) is not None:
+            name += (self.circuit_type_string(meta) + "_")
+        name += "length_{}_seed_{}".format(meta['length_index'], meta['seed'])
+        circuit.name = name
+
+    def circuit_type_string(self, meta: Optional[Dict[str, any]]) -> str:
+        """Returns an additional string describing the exact type of the circuit
+        for non-standard RB experiment; meant to be overridden by derived classes
+        Args:
+            meta: The metadata of the circuit
+        """
+        # pylint: disable=unused-argument
+        return None
+
+    def generate_circuits_for_seed(self) -> List[QuantumCircuit]:
+        """Generates the RB circuits for a single seed
+        Returns:
+            The list of circuits.
+        Additional information:
+            Generates for each seed the sequence C1, C2, ..., Cn of circuits
+            corresponding to the `lengths` parameter.
+
+            Meant to be overriden by derived classes
+        """
+        element_list = self.generate_random_element_list(self._lengths[-1])
+        element_lists = self.split_element_list(element_list, self._lengths)
+        circuits = self.generate_circuits_from_elements(element_lists)
+        return circuits
+
+    def generate_circuits_from_elements(self, element_lists: List) -> List[QuantumCircuit]:
+        """Generates the circuits corresponding to the group elements
+        Args:
+            element_lists: Lists of group elements to build the circuits from
+        Returns:
+            A list of circuits.
+        Additional information:
+            If the element lists are L1, L2, ..., Ln then the resulting circuits
+            are obtained from the gates in L1, L1+L2, L1+L2+L3, ... L1+...+Ln
+            With an additional inverting gate and measurement at the end of each circuit
+        """
+        result = []
+        qr = QuantumRegister(self.num_qubits(), 'qr')
+        cr = ClassicalRegister(self.num_qubits(), 'cr')
+        circ = QuantumCircuit(qr, cr)
+        current_element = self._rb_group.iden(self.num_qubits())
+        for length_index, element_list in enumerate(element_lists):
+            for element in element_list:
+                current_element = self._rb_group.compose(current_element, element['group_element'])
```

Comment:
All this functionlity is no longer needed, since it is handled by the quantum info operators for clifford and cnot dihredral